match/lib/match/spaceship_ensure.rb

```diff
@@ -45,8 +45,8 @@ def bundle_identifier_exists(username: nil, app_identifier: nil)
       UI.user_error!("Couldn't find bundle identifier '#{app_identifier}' for the user '#{username}'")
     end
 
-    def certificate_exists(username: nil, certificate_id: nil)
-      found = Spaceship.certificate.all.find do |cert|
+    def certificate_exists(username: nil, certificate_id: nil, platform: nil)
+      found = Spaceship.certificate.all(mac: platform == "macos").find do |cert|
```

Comment:
There is a Certificate.find https://github.com/fastlane/fastlane/blob/master/spaceship/lib/spaceship/portal/certificate.rb#L290
But that still calls .all.find.

We should accept a `certificate_ids` list and check them all in 1 pass instead of making N requests containing the same list every time.